clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -338,6 +338,18 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
     if (type_is_instr(memref.instr.type) ||
         memref.instr.type == TRACE_TYPE_PREFETCH_INSTR ||
         memref.instr.type == TRACE_TYPE_INSTR_NO_FETCH) {
+        // We do not bother to support legacy traces without encodings.
```

Comment:
This comment is not relevant here. It should be placed near the invariant check where we use `has_encoding`.